### DISCUSSION TITLE
Contributor meeting September 2023

### DISCUSSION THREAD

**Discussion by Aggror** [stride3d/stride#1777] [Contributors]
I would like to have a Stride engine contributor meeting (purely for the Stride repo contributors for now to keep clutter to a minimum)

- This could be a Discord meeting. Since time zone is a difficulty, we can also just use a dedicated discord channel for contributors, 

Topics (WIP)
- [ ] New website summary
- [ ] New documentation: release and process https://github.com/stride3d/stride/discussions/1745
- [ ] Build pipelines and releases (stride engine, docs, website): how does it work now and who can do/trigger what? 
- [ ] New videotutorials
- [ ] Ongoing work from members: shader compiler
- [ ] Release 4.2: https://github.com/stride3d/stride/discussions/1699

**Comment by VaclavElias** [comment]
Just some entry comments on these subjects for our meeting from my side:

- **New Website Summary:** The majority of issues in the backlog are "nice-to-have" improvements. Overall, the new website is performing well, with the vast majority of feedback being positive 🌟. In the long term, we need to continue adding blog posts and explore ways to better restructure the website. This could involve aligning it with recent standards and competitors. For instance, adding more pages about specific features, content specifically for contributors, and high-level info on how the engine works. Adding lots of C# snippets would make it clear that we're focused on .NET and C#.
- **New Documentation:** The new documentation is ready for release and [staging is live](https://stride-doc-staging.azurewebsites.net/latest/en/) 🚀. After the release, it would be beneficial to review every single manual page, make improvements or corrections, and add more C# code snippets. Additional manual pages should also be added.
  - I plan to shift my focus to the Stride wiki, aiming for a complete overhaul. High-level information targeting contributors could be incorporated into the manual, linking back to our wiki. Godot, for instance, has a section for contributors in its manual. This approach would have the added benefit of being more publicly accessible compared to being hidden in a GitHub Wiki.
- **Build Pipelines and Releases:** Stride Docs and Stride Web are now deployed through GitHub Actions 🛠️. Deployments are triggered **automatically** upon pushes to the `release` or `staging` branches and can also be initiated manually.
  - All works as before, small updates like typos new pages should be safe to do without too much testing, directly through GitHub if needed. The workflow remains the same. Stable things should be in master. We have an option to push to `staging` for the website. Deployment last **4 minutes** for the web and **22 minutes** for docs. There will be a fast track for docs deployment. We should deploy this way:
    -  `master` to `staging` if needed to see and elaborate
    -  `master` to `release` once ready for the release
  - I think, collaborators should be able to trigger the deployment also manually, but if they can merge to `staging` or `release` the deployment will start automatically. I can do a quick demo on this for us if needed for anyone.  
- **New Video Tutorials:** I suggest creating a new comprehensive 2D tutorial 🎥. The final output could be incorporated into our example projects. The tutorial creator should be compensated through Open Collective. While visual aesthetics are not a priority, the tutorial should cover all 2D concepts achievable in Stride.
- **Ongoing Work from Members:** I have no comments here, just kudos to everyone for your hard work 👏.
- **Release 4.2:** The focus should be on testing all existing example projects and enhancing the user experience 🧪. The community should be able to launch and run anything without needing to reach out for help on Discord. We should aim for release ASAP after .NET 8 is out.
  - 4.2 Beta would be welcome for testing purpose
- [**Stride Community Toolkit:**](https://www.vaclavelias.com/stride-community-toolkit/) I'd like to propose this new section. The idea is to create numerous mini blog posts featuring code-only examples 📝 or any other content from the toolkit. While I could use only my personal website for these posts, they would be more beneficial if hosted directly on the Stride website. Would it make sense to create a new, separate GitHub team specifically for the Community Toolkit, inviting additional contributors who works with the toolkit already? I would then transfer ownership to stride3d, and set up a GitHub Action for NuGet releases.
  - The preview release could be part of the 4.2 release as an extra bonus

The last days were tense and unexpected but also cool on our Discord. Thanks everyone.

**Reply by VaclavElias** [reply]
Just some addition:

**New Documentation:** - Anyone updating or modifying the Stride codebase should also focus on enhancing and validating the [XML comments](https://dotnet.github.io/docfx/docs/basic-concepts.html). These comments are crucial for the API Docs 📚. Additionally, we now have the capability to reference extended remarks and examples in an external file. This avoids cluttering the codebase.

**Reply by VaclavElias** [reply]
**Bounties:** Another important topic to consider is bounties 💰. With over 400 new members, it's crucial to identify tasks that can be undertaken. Contributors should have the opportunity to be compensated from our Open Collective funds.

**Reply by Doprez** [reply]
I think there is a huge opportunity here to improve the mobile experience as well. https://discord.com/channels/500285081265635328/1152588396288544819/1154141500859490354 Keepsie has offered some financial support on this goal and I dont think we should let that go to waste. I dont use mobile at all but the game dev community that would be effected by this could be massive and entice more mobile companies to help contribute.

**Comment by raphaelschmitz00** [comment]
I understand about the timezones, but is there any indication of a peak time or something like that?

**Comment by manio143** [comment]
My thoughts before the meeting:
**Build pipelines and releases**:
I think it's important to get the pipelines for PRs into a state where we can easily see if the PR is passing tests or not by looking at the marker. Currently the UWP build is always failing and should be removed.
We should consider monthly or bi-weekly package release (depending on volume of changes). Either maintainers should be able to trigger package update, or approve one that fired from a scheduled trigger.
**Ongoing work**:
I haven't done much for Stride in the past months but I will try to get back to it a bit.
I will continue PR reviews and my focus is on the Plugin system and Roslyn analyzers/generators.
**Bugs**:
Given an increasing number of filed bugs would be good to classify them again. I believe Eideren used to manually do it, but that's too much work. Can we use some bot to auto-detect areas label based on bug content?
Furthermore, it would be good to classify bug impact:
* _Critical_ (prevents engine to be usable for most users),
* _Major_ (prevents most users from using part of the engine),
* _Regular_ (impacts some users in part of the engine and cannot be worked-around),
* _Minor_ (impacts very few users or can be worked around without too much hassle).

**Documentation**:
User manuals are very important, but should focus on high level tasks users deal with every day.
For the majority of the engine we should look to instead document the APIs and describe their design for power users to reference.
I think leveraging the community for small contained code examples is key, while contributors should focus on making it easier to understand the complex and undocumented parts of the engine.

Shout out to Vašo for the amazing work on the website and contributors wiki - I had zero issues contributing to the website thanks to the comprehensive wiki page.

P.S. I'm a little overwhelmed with how much is suddenly happening on the daily basis and thus I will try to keep my focus unless tagged for questions and discussions. Big shout out to all of you guys who manage to be active and help people out!

**Reply by xen2** [reply]
Sorry. coincidentally I removed the UWP build just 2 hours ago!
